x-pack/test_serverless/api_integration/test_suites/common/alerting/helpers/alerting_wait_for_helpers.ts

```diff
@@ -301,3 +303,36 @@ export async function waitForEventLog({
     { retries: 10 }
   );
 }
+
+export async function waitForNumRuleRuns({
+  supertest,
+  numOfRuns,
+  ruleId,
+  esClient,
+  testStart,
+}: {
+  supertest: SuperTest<Test>;
+  numOfRuns: number;
+  ruleId: string;
+  esClient: Client;
+  testStart: Date;
+}) {
+  for (let i = 0; i < numOfRuns; i++) {
+    await pRetry(
+      async () => {
+        const resp = await runRule({ supertest, ruleId });
+        if (resp.status !== 204) {
+          throw new Error(`Expected ${resp.status} to equal 204`);
+        }
+        await waitForEventLog({
```

Comment:
I'm a little worried about this function.  It doesn't seem to filter on the rule id, so will match any rule currently running - that we might have forgotten to delete/disable - so could go flaky.  Also the name should indicate it's specifically looking for execution event log docs, since there are many different kinds.  I think we use this function a lot, any chance we can just it here as well - or I guess copy it in if the module itself can't be used here?  

https://github.com/elastic/kibana/blob/main/x-pack/test/alerting_api_integration/common/lib/get_event_log.ts

Reply:
Sure, I can update it to filter on the rule id and try to make it more flake proof

Reply:
Thanks for catching! Resolved in this commit https://github.com/elastic/kibana/pull/164091/commits/18c0755ecc8cedced93b83e2b2573e55249080ed